core/src/main/java/org/apache/spark/shuffle/sort/ShuffleExternalSorter.java

```diff
@@ -255,6 +253,10 @@ public long spill(long size, MemoryConsumer trigger) throws IOException {
 
     writeSortedFile(false);
     final long spillSize = freeMemory();
+    inMemSorter.reset();
+    // Reset the in-memory sorter's pointer array only after freeing up the memory pages holding the
```

Comment:
IMO, comment in `ShuffleExternalSorter` makes it easier to get the context and understand. Also in future, if some one tries to move this call, he will not do so, seeing the comment. If the comment is in the `reset()` function, someone might inadvertently move this call without seeing the comment in `reset()` function.  However, if you have a strong opinion about it, I would gladly move the comment into `reset()`. Let me know what you think.